Sources/MongoSwift/WriteConcern.swift

```diff
@@ -131,17 +122,42 @@ public class WriteConcern: Codable {
         }
     }
 
-    /// Initializes a new `WriteConcern` by copying a `mongoc_write_concern_t`.
+    /// Initializes a new `WriteConcern` with the same values as the provided `mongoc_write_concern_t`.
     /// The caller is responsible for freeing the original `mongoc_write_concern_t`.
     internal init(from writeConcern: OpaquePointer?) {
-        self._writeConcern = mongoc_write_concern_copy(writeConcern)
+        if mongoc_write_concern_journal_is_set(writeConcern) {
+            self.journal = mongoc_write_concern_get_journal(writeConcern)
+        } else {
+            self.journal = nil
+        }
+
+       let number = mongoc_write_concern_get_w(writeConcern)
+        switch number {
+        case MONGOC_WRITE_CONCERN_W_DEFAULT:
+            self.w = nil
+        case MONGOC_WRITE_CONCERN_W_MAJORITY:
+            self.w = .majority
+        case MONGOC_WRITE_CONCERN_W_TAG:
+            if let wTag = mongoc_write_concern_get_wtag(writeConcern) {
+                self.w = .tag(String(cString: wTag))
+            }
+        default:
+            self.w = .number(number)
+        }
+
+       let wtimeout = Int64(mongoc_write_concern_get_wtimeout(writeConcern))
+       if wtimeout == 0 {
+           self.wtimeoutMS = nil
+       } else {
+           self.wtimeoutMS = wtimeout
+       }
     }
 
     private enum CodingKeys: String, CodingKey {
         case w, j, wtimeout
     }
 
-    public required convenience init(from decoder: Decoder) throws {
+    public init(from decoder: Decoder) throws {
```

Comment:
rather than having this manual `decode` implementation, I think we can delete this and use the synthesized one now that we store all the properties.

*but*, in order for the synthesized one to be correct, we probably need to tell the compiler about how the key names don't quite match up. WriteConcern documents use `j` for `journal` and `wtimeout` for `wtimeoutMS`. 

we can do that by updating the `CodingKeys` to reflect this:
```swift
private enum CodingKeys: String, CodingKey {
   case w, journal = "j", wtimeoutMS = "wtimeout"
}
```
This tells the compiler that the value for our `w` property will be stored in a document under the key `"w"`, but that `journal` will be stored under `"j"` and `wtimeoutMS` will be stored under `"wtimeout"`.

you will then need to also update the `encode` method to use those new CodingKey cases, e.g. the last line becomes `try container.encodeIfPresent(self.journal, forKey: .journal)`